### PR TITLE
Add impl for shopping cart

### PR DESCRIPTION
**Pull request by VdBondarev**

### CODE REVIEW

File: src/main/java/mate/academy/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart controller", description = "Endpoints for managing shopping carts")
+@RequestMapping(value = "/api/cart")
```
Comment (okuzan) [inline src/main/java/mate/academy/controller/ShoppingCartController.java:23]: move /api to server.servlet.context-path in properties

File: src/main/java/mate/academy/mapper/CartItemMapper.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package mate.academy.mapper;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.dto.cart.item.CartItemResponseDto;
+import mate.academy.model.ShoppingCart;
+
+public class CartItemMapper {
+    private CartItemMapper() {
+    }
```
Comment (okuzan) [inline src/main/java/mate/academy/mapper/CartItemMapper.java:10]: use mapstruct for mapping
use default @AfterMapping methods if needed

File: src/main/java/mate/academy/service/shopping/cart/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                .findFirst();
+        if (item.isEmpty()) {
+            CartItem cartItem = new CartItem();
+            cartItem.setQuantity(quantity);
+            cartItem.setShoppingCart(shoppingCart);
+            cartItem.setBook(book);
+            Set<CartItem> cartItems = shoppingCart.getCartItems();
+            cartItems.add(cartItemRepository.save(cartItem));
+            shoppingCart.setCartItems(cartItems);
+            return shoppingCartMapper.toDto(shoppingCartRepository.save(shoppingCart));
```
Comment (okuzan) [inline src/main/java/mate/academy/service/shopping/cart/ShoppingCartServiceImpl.java:47]: try to delegate some of this logic to private methods, method looks bulky

**Review by okuzan — CHANGES_REQUESTED** [review]
GJ

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**